src/v/cluster/id_allocator_stm.cc

```diff
@@ -166,19 +166,29 @@ ss::future<> id_allocator_stm::apply(model::record_batch b) {
 
         _processed++;
         if (_processed > _log_capacity) {
-            return _c
-              ->write_snapshot(
-                raft::write_snapshot_cfg(_next_snapshot, iobuf()))
-              .then([this] {
-                  _next_snapshot = _insync_offset;
-                  _processed = 0;
-              });
+            (void)ss::with_gate(_gate, [this] { return write_snapshot(); });
```

Comment:
@dotnwat this is the only change I've done compared to the original commit: `ssx:spawn_with_gate` to `(void)ss::with_gate`